### ISSUE TITLE
Compiling PJS fails with NPE

### ISSUE THREAD

**Issue by cramforce** [google/closure-compiler#2160]
File https://github.com/jridgewell/PJs/blob/master/promise.js#L278

Relevant code 
```JS
function Deferred(Promise) {
  var deferred = this;
  /** @type {!Promise} */
  this.promise = new Promise(function(resolve, reject) {
    /** @type {function(*=)} */
    deferred.resolve = resolve;

    /** @type {function(*=)} */
    deferred.reject = reject;
  });
  return deferred;
}
```

Error

```
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(NAME Promise): node_modules/promise-pjs/promise.js:278:21
  this.promise = new Promise(function(resolve, reject) {
  Parent(NEW): node_modules/promise-pjs/promise.js:278:17
  this.promise = new Promise(function(resolve, reject) {

	at com.google.javascript.jscomp.NodeUtil.getRValueOfLValue(NodeUtil.java:4365)
	at com.google.javascript.jscomp.ProcessCommonJSModules$RewriteModule.getModuleImportName(ProcessCommonJSModules.java:1065)
	at com.google.javascript.jscomp.ProcessCommonJSModules$RewriteModule.maybeUpdateName(ProcessCommonJSModules.java:761)
	at com.google.javascript.jscomp.ProcessCommonJSModules$RewriteModule.visit(ProcessCommonJSModules.java:568)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:642)
	at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:714)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:638)
	at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:714)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:638)
	at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:714)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:638)
	at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:714)
	at com.google.javascript.jscomp.NodeTraversal.traverseBlockScope(NodeTraversal.java:729)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:636)
	at com.google.javascript.jscomp.NodeTraversal.traverseFunction(NodeTraversal.java:673)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:630)
	at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:714)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:638)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:326)
	at com.google.javascript.jscomp.NodeTraversal.traverseEs6(NodeTraversal.java:588)
	at com.google.javascript.jscomp.ProcessCommonJSModules.process(ProcessCommonJSModules.java:124)
```

**Comment by ChadKillingsworth** [comment]
Was that from HEAD?

**Comment by cramforce** [comment]
Used `"google-closure-compiler": "^20161024.3.0",`

**Comment by cramforce** [comment]
It may be a recent regression. `v20160911 (Sep 11, 2016)` (Used by AMP) successfully compiles the file.

**Comment by ChadKillingsworth** [comment]
It's definitely a regression - I basically rewrote the entire pass between those releases.

**Comment by ChadKillingsworth** [comment]
Closed by c4a104075d40dd2191b45de33d00b79edd45b97e
